bot/exts/nominations.py

```diff
@@ -0,0 +1,81 @@
+import re
+
+import discord
+from discord.ext import commands
+
+from bot import constants, logger
+from bot.bot import ThreadBot
+from bot.utils.helpers import get_thread_from_message_id
+
+NOMINATION_MESSAGE_REGEX = re.compile(
+    r"<@!?\d+> \((.+)#\d{4}\) for Helper!\n\n\*\*Nominated by:\*\*",
+    re.MULTILINE
+)
+
+# When nominations are posted manually, the Discord message box converts +1/-1 to the thumbs emojis
+NOMINATION_ENDING_TEXT = "react 👍 for approval, or 👎 for disapproval*."
+
+
+class Nominations(commands.Cog):
+    """Cog for getting information about PyPi packages."""
+
+    def __init__(self, bot: ThreadBot):
+        self.bot = bot
+        # Stores the name of the member recently posted nomination as,
+        # nomiantion messages can be split into multiple messages.
+        self.nominated_member_name: int = None
+
+        if constants.DEBUG_MODE:
+            self.archive_time = constants.ThreadArchiveTimes.DAY.value
+        else:
+            self.archive_time = constants.ThreadArchiveTimes.WEEK.value
+
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message) -> None:
+        """Creates a thread whenever a vote is sent in the nominations voting channel."""
+        if message.channel.id != constants.Channels.nomination_voting:
+            return  # Ignore messages not in voting channel.
+
+        if match := NOMINATION_MESSAGE_REGEX.match(message.content):
+            self.nominated_member_name = match.group(1)
+
+        if not message.content.endswith(NOMINATION_ENDING_TEXT):
+            # Only create thread on final message, as a vote could be split into multiple messages.
+            return
```

Comment:
If a vote message is too long (due to very long nomination reasons from multiple people) the bot posts the vote over 2 or more messages. This logic is here so that we get the name of the nominated member from the first message, then wait for the last message to be detected, and make a thread on that.

I don't think we need to worry about 2 nominations coming in at the same time causing an issue here, as its *extremely* unlikely to get conditions that allow for that.

Reply:
Then I think for safety that before setting a new name that you have detected, check that it it is None and do `logger.error()` if not.

Reply:
Then I think for safety that before setting a new name that you have detected, check that it it is None and do `logger.error()` if not.